lib/balrog/middleware.rb

```diff
@@ -7,35 +7,58 @@
 # This is typically set up in an initialize when you run
```

Comment:
This file is getting quite chunky now. I wonder if we can split it up a bit? How about lumping all the handlers (login, omniauth, and logout) together in a module which could just be `include`d in here? e.g. have a `Balrog::Middleware::Handlers` module with those methods in.